From 1a7b56b8a5f6c5d019639c7c6da4241e9c738084 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 23 Jun 2021 19:46:48 +0300
Subject: [PATCH] [SW-46517] greco: set boot restart bits

This patch configures boot restart register and enable bits
for resets for which boot sequence should be started.

Change-Id: I3f15003a038283c9bd105a815384b72a0c332ce2
---

diff --git a/arch/greco_arc/common.c b/arch/greco_arc/common.c
index 07a4edf..1ae5b23 100644
--- a/arch/greco_arc/common.c
+++ b/arch/greco_arc/common.c
@@ -324,6 +324,12 @@
 {
 	/* Functionality used to be done in BTL */
 	init_prstn();
+
+	/*
+	 * Upon MANUAL, WATCHDOG, SW ALL, PRSTN, FLR and FW reset
+	 * restart boot sequence.
+	 */
+	regmap_writel(0x770, mmPSOC_GLOBAL_CONF_BOOT_SEQ_RE_START);
 }
 
 static void halt_n_reset_arc(u8 id)
